src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -650,7 +652,26 @@ public BulkResponse executeBulk(BulkRequest bulk) throws IOException {
 
     final String errorInfo = errors.isEmpty() ? result.getErrorMessage() : errors.toString();
 
-    return BulkResponse.failure(retriable, errorInfo);
+    Map<IndexableRecord, BulkResultItem> failedResponses = new HashMap<>();
+    List<BulkResultItem> items = result.getItems();
```

Comment:
I'd also suggest declaring and using below (example names): 
```
List<IndexableRecord> bulkRecords = jestBulk.records();
int bulkRecordsNum = bulkRecords.size();
int itemsNum = items.size();
```
probably JIT-able optimizations but since not bad to have and maybe slightly easier to follow.